ingestion/functions/parsing/common/python/parsing_lib/parsing_lib.py

```diff
@@ -43,21 +43,13 @@ def write_to_server(cases, headers):
     Upserts the provided cases via the G.h Case API.
 
     TODO: Link out to the Case API resource documentation, once available.
-    TODO: Parallelize these requests.
     """
-    count_success = 0
-    count_error = 0
-    put_api_url = f"{os.environ['SOURCE_API_URL']}/cases"
+    put_api_url = f"{os.environ['SOURCE_API_URL']}/cases/batchUpsert"
     print(f"Sending {len(cases)} cases to {put_api_url}")
-    for case in cases:
-        try:
-            requests.put(put_api_url, json=case,
-                         headers=headers).raise_for_status()
-            count_success += 1
-        except Exception as e:
-            print(e)
-            count_error += 1
-    return count_success, count_error
+    res = requests.post(put_api_url, json={"cases": cases},
+                       headers=headers)
+    res_json = res.json()
+    return len(res_json["createdCaseIds"]), len(res_json["updatedCaseIds"])
```

Comment:
We probably want to surface some indication of the errors, here.

- Error count, as we're currently doing, though the meaning will change (since we could have up to `n` validation errors, but only 1 server error for the single API call).
- Index of cases that produced the errors.
- Raw errors.

WDYT?

Reply:
I'm all for it but I don't see any way to surface granular errors at this point, the curator service just proxies to the data service and the data service batch upsert just looks at what's returned by bulkWrite:
http://mongodb.github.io/node-mongodb-native/3.1/api/Collection.html#~BulkWriteOpResult

that doesn't contain errors or anyting that would make it possible to return what you proposed above, am I missing something?

Reply:
I will merge this and we can talk about errors in a future PR.